cmake/HPX_SetupMPI.cmake

```diff
@@ -18,38 +18,30 @@ if(HPX_WITH_NETWORKING AND HPX_WITH_PARCELPORT_MPI)
     hpx_error("MPI could not be found and HPX_WITH_PARCELPORT_MPI=ON, please specify \
     MPI_ROOT to point to the root of your MPI installation")
   endif()
+
   add_library(hpx::mpi INTERFACE IMPORTED)
   set_property(TARGET hpx::mpi PROPERTY
     INTERFACE_INCLUDE_DIRECTORIES ${MPI_INCLUDE_PATH} ${MPI_CXX_INCLUDE_DIRS})
   # MPI_LIBRARY and EXTRA is deprecated but still linked for older MPI versions
-  if (MPI_CXX_LIBRARIES)
-    if(${CMAKE_VERSION} VERSION_LESS "3.12.0")
-      set_property(TARGET hpx::mpi APPEND PROPERTY
-        INTERFACE_LINK_LIBRARIES ${MPI_CXX_LIBRARIES})
-    else()
-      target_link_libraries(hpx::mpi INTERFACE ${MPI_CXX_LIBRARIES})
-    endif()
-  endif()
-  # Ensure compatibility with older versions
-  if (MPI_LIBRARY)
```

Comment:
Good catch! For others, `MPI_LIBRARY` and `MPI_EXTRA_LIBRARY` have been deprecated since at least CMake 3.5 so we can safely remove these.